src/GdbCommand.h

```diff
@@ -27,28 +28,146 @@ class GdbCommand {
   virtual std::string invoke(GdbServer& gdb_server, Task* t,
                              const std::vector<std::string>& args) = 0;
 
+  /**
+   * When call, gdb will automatically fill argument to this command
+   * by running gdb.execute(). This is useful to gdb state along side
+   * of the command invocation.
+   */
+  void add_auto_args(const std::string& auto_arg) {
+    cmd_auto_args.push_back(auto_arg);
+  }
+
+  const std::vector<std::string>& auto_args() const { return cmd_auto_args; }
+
+  /**
+   * Setup all the automatic auto_args for our commands.
+   */
+  static void init_auto_args();
+
 private:
   const std::string cmd_name;
+  std::vector<std::string> cmd_auto_args;
 };
 
 #define RR_LINE_CONCAT(str, line) str##line
 #define RR_LINE_EXPAND(str, line) RR_LINE_CONCAT(str, line)
-#define RR_CMD_CLASSNAME() RR_LINE_EXPAND(RRCmd, __LINE__)
 #define RR_CMD_OBJ() RR_LINE_EXPAND(sRRCmdObj, __LINE__)
 
-#define RR_CMD(name)                                                           \
-  class RR_CMD_CLASSNAME() : public GdbCommand {                               \
+#define RR_CMD(id, name)                                                       \
+  class id : public GdbCommand {                                               \
+  public:                                                                      \
+    id() : GdbCommand(name) {}                                                 \
+                                                                               \
+  private:                                                                     \
+    virtual std::string invoke(GdbServer& gdb_server, Task* t,                 \
+                               const std::vector<std::string>& args);          \
```

Comment:
Yes, so that point is a wash for either solution.

Anyways it's not important to the PR so I'm just going to split out these parts and rebase on top of what's already landed.